# workaround for:
# config.sh: line 17: warning: here-document at line 17 delimited by end-of-file (wanted `EOF')
# config.sh: line 16: warning: here-document at line 16 delimited by end-of-file (wanted `EOF')
# config.sh: line 18: syntax error near unexpected token `('
# make: *** [config.vars] Error 2

diff --git a/config.sh b/config.sh
index 4b32da2..459b1be 100755
--- a/config.sh
+++ b/config.sh
@@ -13,20 +13,11 @@ XUL_VERSION=$(echo '#include "mozilla-config.h"'|
 XUL_VER_MIN=$(echo $XUL_VERSION | sed -r -e 's/([^.]+\.[^.]+).*/\1/g')
 XUL_VER_MAX=$(echo $XUL_VERSION | sed -rn -e 's/([^.]+).*/\1.*/gp')
 
-HAVE_NSILMS_GETISLOGGEDIN=$({ cat <<EOF; } | $CXX $XUL_CFLAGS $GNOME_CFLAGS $CXXFLAGS -x c++ -w -c -o /dev/null - 2>/dev/null && echo 1 || echo 0)
-#include "$SRC_GNOME_KEYRING_H"
-NS_IMETHODIMP GnomeKeyring::GetIsLoggedIn(bool *aIsLoggedIn) { return NS_OK; }
-EOF
-
-HAVE_NSILMS_INITWITHFILE_2=$({ cat <<EOF; } | $CXX $XUL_CFLAGS $GNOME_CFLAGS $CXXFLAGS -x c++ -w -c -o /dev/null - 2>/dev/null && echo 1 || echo 0)
-#include "$SRC_GNOME_KEYRING_H"
-NS_IMETHODIMP GnomeKeyring::InitWithFile(nsIFile *aInputFile, nsIFile *aOutputFile) { return NS_OK; }
-EOF
-
-HAVE_MOZ_BUG_956507=$({ cat <<EOF; } | $CXX $XUL_CFLAGS $GNOME_CFLAGS $CXXFLAGS -x c++ -w -c -o /dev/null - 2>/dev/null && echo 0 || echo 1)
-#include <nspr/prtypes.h>
-#include "mozilla/Char16.h"
-EOF
+HAVE_NSILMS_GETISLOGGEDIN=1
+
+HAVE_NSILMS_INITWITHFILE_2=0
+
+HAVE_MOZ_BUG_956507=1
 
 HAVE_MOZGLUE=$($CXX $XUL_CFLAGS $XUL_LDFLAGS $XPCOM_ABI_FLAGS $CXXFLAGS $LDFLAGS -lmozglue -shared -o /dev/null && echo 1 || echo 0)
 
